### PR TITLE
python-sphinxcontrib-serializinghtml: package upgrade to 1.1.10

### PR DESCRIPTION
**Pull request by amritakohli**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Upgrade python-sphinxcontrib-serializinghtml package to 1.1.10 ahead of Mariner 3.0 release.

###### Change Log  

- Upgraded package.
- Uploaded source tarball. 
- Updated signatures and cgmanifest.
- Updated tests and build commands to use modern pyproject instead of setup.py in line with upstream changes.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Locally built successfully with RUN_CHECK=y 
- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=510472&view=results

### CODE REVIEW

File: SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec [hunk 1]
```diff
@@ -2,22 +2,24 @@
...
-Version:        1.1.5
-Release:        5%{?dist}
+Version:        1.1.10
+Release:        1%{?dist}
 License:        BSD
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 URL:            https://www.sphinx-doc.org/en/master/
-Source0:        https://files.pythonhosted.org/packages/b5/72/835d6fadb9e5d02304cf39b18f93d227cd93abd3c41ebf58e6853eeb1455/%{pypi_name}-%{version}.tar.gz
+Source0:        https://files.pythonhosted.org/packages/54/13/8dd7a7ed9c58e16e20c7f4ce8e4cb6943eb580955236d0c0d00079a73c49/sphinxcontrib_serializinghtml-1.1.10.tar.gz#/%{pypi_name}-%{version}.tar.gz
```
Comment (bfjelds) [inline SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec:11]: nit: why the change away from ${pypi_name} and %{version}?

File: SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec [hunk 2]
```diff
@@ -2,22 +2,24 @@
...
 
 BuildArch:      noarch
 
 BuildRequires:  gettext
 BuildRequires:  python3-devel
 BuildRequires:  python3-setuptools
+BuildRequires:  python%{python3_pkgversion}-pip
```
Comment (bfjelds) [inline SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec:18]: nit: maybe better to be consistent about using python3 or python%{pyton3_pkgversion} for all requires.

File: SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec (hunk 1 above)
Comment (amritakohli) [reply SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec:11]: because of how the tar file is named when unzipped
it's a dash vs underscore thing
%version here won't add anything as the link would change entirely when upgrading since it has a unique hash in it

Comment (bfjelds) [reply SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec:11]: it seems like the version could still be used?

Comment (amritakohli) [reply SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec]: updated to use it

**Review by ypanch — APPROVED** [review]

File: SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec (hunk 1 above)
Comment (bfjelds) [reply SPECS/python-sphinxcontrib-serializinghtml/python-sphinxcontrib-serializinghtml.spec]: i think i would suggest introducing a new variable like %global pypi_name sphinxcontrib-serializinghtml, except with the underscore ... and use that in the Source and the autosetup line.  could be something like:
 
%global pypi_name_prefix sphinxcontrib
%global pypi_name_suffix serializinghtml
%global pypi_name %{pypi_name_prefix}-${pypi_name_suffix}
%global pypi_name_underscore %{pypi_name_prefix}_${pypi_name_suffix}
